optimum/onnxruntime/trainer.py

```diff
@@ -134,6 +139,30 @@
 SCALER_NAME = "scaler.pt"
 
 
+class ModuleWithLoss(nn.Module):
+    def __init__(self, model, args) -> None:
+        super().__init__()
+        self._original_model = model
+        self.args = args
+
+        # Creating an instance of huggingFace Trainer so we can use compute_loss() logic and avoid duplicated code.
+        self.hf_trainer = Trainer(model)
+        # Label smoothing
+        if self.args.label_smoothing_factor != 0:
+            from transformers.trainer_pt_utils import LabelSmoother
+
+            self.label_smoother = LabelSmoother(epsilon=self.args.label_smoothing_factor)
+        else:
+            self.label_smoother = None
+
+    def forward(self, inputs: Dict[str, Union[torch.Tensor, Any]], return_outputs):
+        return self.hf_trainer.compute_loss(self._original_model, inputs, return_outputs=False)
```

Comment:
I think this is not correct. doing this way, self.label_smoother won't be used by the compute_loss. What I suggested earlier is to bind the compute_loss of hf Trainer to ModuleWithLoss. Here is the example:
```

class A:
    def __init__(self) -> None:
        self.prop = "A's prop"

    def f(self, x: int) -> int:
        print("A>>f is called, the prop used is: ", self.prop)
        return x + 1

class B:
    def __init__(self) -> None:
        self.prop = "B's prop"

    def main(self, x: int) -> int:
        print("B>>main is called")
        self.f(x)
        return x + 2

    def f(self, x: int) -> int:
        raise NotImplementedError()


b_instance = B()
import types

b_instance.f = types.MethodType(getattr(A,'f'), b_instance)

b_instance.main(1)

```

Reply:
B>>main is called
A>>f is called, the prop used is:  B's prop

Reply:
FYI @JingyaHuang , In this case the code in the Trainer should be maintained if the compute_loss in hf trainer got changed then it should be changed in the forward pass of ModuleWithLoss as well.

Reply:
Hi @AdamLouly, sorry for the back and forth. I proposed to rewrite the codes as I was considering opening a PR in Transformers to put the label smoother inside forward. If so we don't need a wrapper in Optimum. But as @pengwa explained, a PR in Transformers won't be enough (we can't limit unnecessary outputs in Transformers for the flexibility reason), so we will always need this wrapper in Optimum. 

If so, I agree that we should inherit the `compute_loss()`(as you did before) to ease the maintenance.